R/plumber.R

```diff
@@ -887,9 +887,20 @@ Plumber <- R6Class(
     #' @param api This can be
     #'   * an OpenAPI Specification formatted list object
     #'   * a function that accepts the OpenAPI Specification autogenerated by `plumber` and returns a OpenAPI Specification formatted list object.
+    #'   * a path to an OpenAPI Specification
     #'
     #'  The value returned will not be validated for OAS compatibility.
     setApiSpec = function(api = NULL) {
+      if (is.character(api)) {
+        if (tools::file_ext(api) %in% c("yaml", "yml")) {
+          if (!requireNamespace("yaml", quietly = TRUE)) {
+            stop("yaml must be installed to read yaml format")
+          }
+          api <- yaml::read_yaml(api, eval.expr = FALSE)
+        } else {
+          api <- jsonlite::parse_json(api, simplifyVector = TRUE)
```

Comment:
```suggestion
          api <- jsonlite::read_json(api, simplifyVector = TRUE)
```

Reply:
Same reason I went with parse_json, to allow for URL, read_json only work from disk files. If we do not need to support remote API file, I'm all for it.